deployment/src/main/java/org/wso2/testgrid/deployment/deployers/ShellDeployer.java

```diff
@@ -50,29 +52,60 @@ public String getDeployerName() {
         try {
+            Script deployment = getScriptToExecute(testPlan.getDeploymentConfig(), Script.Phase.CREATE);
+            logger.info("Performing the Deployment " + deployment.getName());
+            String infraArtifact = StringUtil
+                    .concatStrings(infrastructureProvisionResult.getResultLocation(),
+                            File.separator, "k8s.properties");
+            StringBuilder builder = new StringBuilder();
+            builder.append("$(cat ").append(infraArtifact).append(") ");
+
+            deployment.getInputParameters().forEach((o, o2) -> {
+                builder.append(o2).append(" ");
+            });
+
             ShellExecutor shellExecutor = new ShellExecutor(Paths.get(TestGridUtil.getTestGridHomePath()));
-            // TODO: DEPLOY_SCRIPT_NAME is hard-coded. We should first find the script name in the DeploymentConfig
             if (!shellExecutor
-                    .executeCommand("bash " + Paths
-                            .get(infrastructureProvisionResult.getDeploymentScriptsDir(), DEPLOY_SCRIPT_NAME))) {
+                    .executeCommand("bash " + Paths.get(infrastructureProvisionResult
+                            .getDeploymentScriptsDir(), deployment.getFile()) + " " + builder.toString())) {
                 throw new TestGridDeployerException("Error occurred while executing the deploy script");
             }
         } catch (CommandExecutionException e) {
             throw new TestGridDeployerException(e);
         } catch (IOException e) {
             throw new TestGridDeployerException("Error occurred while retrieving the Testgrid_home ", e);
         }
-
         DeploymentCreationResult result = new DeploymentCreationResult();
         result.setName(deploymentPatternConfig.getName());
         result.setHosts(infrastructureProvisionResult.getHosts());
         result.setDeploymentScriptsDir(infrastructureProvisionResult.getDeploymentScriptsDir());
-
         return result;
     }
+
+    /**
+     * This method returns the script matching the correct script phase.
+     *
+     * @param deploymentConfig {@link DeploymentConfig} object with current deployment configurations
+     * @param scriptPhase {@link Script.Phase} enum value for required script
+     * @return the matching script from deployment configuration
+     * @throws TestGridDeployerException if there is no matching script for phase defined
+     */
+    private Script getScriptToExecute(DeploymentConfig deploymentConfig, Script.Phase scriptPhase)
+            throws TestGridDeployerException {
+
+        for (Script script : deploymentConfig.getDeploymentPatterns().get(0).getScripts()) {
+            if (scriptPhase.equals(script.getPhase())) {
+                return script;
+            }
+        }
+        throw new TestGridDeployerException("The Script list Provided doesn't containt a " + scriptPhase.toString() +
```

Comment:
If phase is empty, shall we return the very first script instead of failing the build? I think in the default case (ie. deployment creation) we can keep the phase empty.

Reply:
Won't that result in unexpected behaviors since any script can be returned?